client/blocks/comments/block-editor/index.jsx

```diff
@@ -28,16 +28,27 @@ const allowedBlocks = [
 ];
 
 const BlockEditor = ( { onChange, suggestions } ) => {
+	const [ isLoaded, setIsLoaded ] = useState( false );
+
 	useEffect( () => {
-		getAddAutocompleters( suggestions ).then( ( addAutoCompleters ) => {
-			addFilter(
-				'editor.Autocomplete.completers',
-				'readerComments/autocompleters',
-				addAutoCompleters
-			);
-		} );
+		// ensure that the addAutoCompleters filter is added before the IsolatedBlockEditor is loaded
+		// so that the filters are definitely run
+		new Promise( ( resolve ) => {
+			getAddAutocompleters( suggestions ).then( ( addAutoCompleters ) => {
```

Comment:
why do we need to put this in a promise? it seems like:

```js
useEffect( async () => {
  const addAutocompleters = await getAddAutocompleters( suggestions );
  addFilter(
    'editor.Autocomplete.completers',
    'readerComments/autocompleters',
    addAutoCompleters
  );
  setIsLoaded( true );
}, [ suggestions ] );
```

is equivalemt, but maybe I'm missing something? Also, do we need to make sure this only runs once, instead of whenever suggestions changes? 

But yeah, this seems like a fair approach. I'm not super familiar with the problem space, but I don't immediately see a different way to handle it.

Reply:
React won't let you pass an async function (or anything else that returns a promise) to `useEffect` but yes, it seems that `addFilter` and `setIsLoaded` could be in one `then` callback and that `new Promise` is not needed.

Reply:
Definitely, my bad, I'll make that change.